pkg/nfd-master/nfd-master.go

```diff
@@ -605,19 +605,74 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(nodeName string) error {
 	return nil
 }
 
-func (m *nfdMaster) refreshNodeFeatures(cli *kubernetes.Clientset, nodeName string, annotations, labels map[string]string, features *nfdv1alpha1.Features) error {
+// validateExtendedResources validates extended resources and returns a map
+// of valid extended resources.
+func (m *nfdMaster) validateExtendedResources(features *nfdv1alpha1.Features, resourceLabels map[string]string) map[string]string {
```

Comment:
nit: the name resourceLabels is very misleading here IMO

```suggestion
func (m *nfdMaster) validateExtendedResources(features *nfdv1alpha1.Features, extendedResources map[string]string) map[string]string {
```